SETUP.md

```diff
@@ -0,0 +1,78 @@
+
+```
+refd_pkgs <- 
+  c('animation', 'blastula', 'blogdown', 'bookdown','broom', 'Cairo', 'cairoDevice', 'condformat',
+    'dagitty', 'diagram', 'DiagrammeR', 'distill', 'downloadthis', 'DT', 'ezknitr', 'flair',
+    'flexdashboard', 'flextable', 'formatR', 'formattable', 'gganimate', 'ggdag', 'ggplot2', 
+    'gifski', 'googledrive', 'govdown', 'gt', 'gtsummary', 'here', 'huxtable', 'kableExtra', 
+    'knitcitations', 'knitr', 'learnr', 'magick', 'nomnoml', 'officedown', 'officer', 'pagedown',
+    'pander', 'pixiedust', 'pkgdown', 'printr', 'r2d3', 'reactable', 'reticulate', 'revealjs', 'rgl',
+    'rhandsontable', 'rmarkdown', 'rmdformats', 'roxygen2', 'rsconnect', 'rticles', 'sass', 'spelling', 
+    'stargazer', 'styler', 'svglite', 'tables', 'tangram', 'tikzDevice', 'tinytex', 'tufte', 'usethis',
+    'webshot', 'workflowr', 'xaringan', 'xfun', 'xtable', 'yaml', 'ztable')
+```
+
+You may install packages with `install.packages()` or with the **pak** package's `pkg_install()` function, i.e.:
+
+```
+pak::pkg_install(reqd_pkgs)
+```
+
+Note the `pkg_install()` will only install missing packages (and not update current packages) unless you set `upgrade = TRUE`. It is generally a good idea to work with the latest version of packages when preparing a contribution or, at minimum if you receive errors while building, upgrading your packages is a good first step in debugging.
+
+### LaTeX Packages
+
+Due to the cookbook's discussion of various [graphical devices](https://bookdown.org/yihui/rmarkdown-cookbook/graphical-device.html), a number of LaTeX packages are also required. Some may already be installed in `tinytex` depending on what version of the package you have.
+
+If you have never used LaTeX or `tinytex` before, first run:
+
+```
+tinytex::install_tinytex()
+```
+
+To check if you have the needed LaTeX packages, run:
+
+```
+tex_pkgs <- c('pgf', 'preview', 'xcolor', 'grfext')
+setNames(tex_pkgs %in% tinytex::tl_pkgs(), pkgs)
+```
+
+If the results for any of the preceding packages is `FALSE`, you may install it with `tinytex::tlmgr_install()`. 
+
+More about `tinytex` and troubleshooting missing LaTeX packages can be found in the cookbook [here](https://bookdown.org/yihui/rmarkdown-cookbook/install-latex-pkgs.html). 
+
+### Other System Dependencies
+
+For Windows users, the section of the book on the [Asymptote language engine](https://bookdown.org/yihui/rmarkdown-cookbook/eng-asy.html) does not build properly so it will be skipped. For users with other operating systems, this section will be skipped if Asymptote is not installed on your computer. If they wish to build this section, they must download and install the [Asymptote software](https://asymptote.sourceforge.io/) from its website. After you install it, you may confirm that it is on your system's path by running the terminal command `where asy`.
+
+## Building the Book
```

Comment:
I really like this idea. One potential issue is that the majority of chapters call the `import_example()` function which is defined in `index.Rmd`, so attempting to knit a single chapter or using `bookdown::preview_chapter()` throws an error. 

Shall I pull that function into a stand-alone `.R` script and `source()` it in at the beginning of the chapters that need it?

Reply:
Oh I thought `preview_chapter()` was at least running `index.Rmd`... Currently you are saying are book does not work when one wants to preview chapter ? I did not encounter that before 🤔 I see we are using knit and merge approach, so it would have thought it works. I'll check my assumption in bookdown code.

So that would mean some of our content in `index.Rmd` should leave in something `common.R`, and used in `before_chapter_script` option (it is the same as having a chunk sourcing the script I think)

Reply:
For me, with the CRAN version of `bookdown` (0.21), I get the following

```
bookdown::preview_chapter("03-basics.Rmd")
> Quitting from lines 108-135 (rmarkdown-cookbook.Rmd) 
> Error in import_example("spin.R") : 
>  could not find function "import_example"
```

So, it does not seem to be running `index.Rmd` for me. Does that function work for you?

If does work if I pull out the function into its own script and run the `before_chapter_script` option to the YAML as you mention. (Thanks for the tip! Much nicer than `source()`ing it directly.)

Reply:
I believe Yihui did not preview anything when he writes the book... I'll test it but I am sure you're right. 
Do you want to open an issue about that or do we fix it in this PR ?

Reply:
Since the Asymptote change is in this PR, I think we can just put it in this PR, also, as a set of backend changes if that sounds good.

Reply:
You're right ! Thanks.